token/filetoken/filetoken.go

```diff
@@ -0,0 +1,114 @@
+package filetoken
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/user"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/puppetlabs/pe-sdk-go/log"
+	"github.com/puppetlabs/pe-sdk-go/token"
+)
+
+// fileToken struct
+type fileToken struct {
+	path string
+}
+
+// NewFileToken constructs a new filetoken
+func NewFileToken(path string) token.Token {
+	fileToken := fileToken{path: getPath(path)}
+	return &fileToken
+}
+
+// Read reads the token from a file and returns a valid token.
+// If the token is not valid an error will be returned
+func (ft *fileToken) Read() (string, error) {
+	return ft.readToken()
+}
+
+//DeleteFile deletes the token file
+//If the token is not valid, or file is a symlink an error will be returned
+func (ft *fileToken) Delete() error {
+	_, err := ft.readToken()
+	if err != nil {
+		return err
+	}
```

Comment:
`readToken` will return an error if `token` is not valid, in that case we won't delete the file, is this expected?

Reply:
This function mimics the C++ version of the delete-token-file subcommand. For reference here is the function in C++ https://github.com/puppetlabs/puppet-access/blob/main/exe/puppet-access.cc#L326-L331
The reasoning behind this is still unclear to me, but it might be because, if the token is invalid it means it hasn't been generated by the RBAC API, so we shouldn't mess with it.

Reply:
we should test what c++ version does and implement the same

Reply:
I have tested the C++ version, and if the token is not valid, it returns an error saying `Malformed token: token does not match expected formats`.

Reply:
we could add a `--force` option to remove the file no matter the token validity. this could be done now or later

Reply:
@BobosilaVictor can we keep the same error message as the C++ version?